pypsa/plot.py

```diff
@@ -1233,3 +1243,160 @@ def iplot(
             pltly.iplot(fig)
 
     return fig
+
+
+def explore(
+    n,
+    crs: str = "EPSG:4326",
+    tooltip=True,
+    popup=True,
+):
+    """
+    Create an interactive map displaying PyPSA network components using geopandas exlore() and folium.
+
+    This function generates a Folium map showing buses, lines, links, and transformers from the provided network object.
+
+    Parameters
+    ----------
+    n : object
+        PyPSA.Network object containing components `buses`, `links`, `lines`, and `transformers`.
+    crs : str, optional, default="EPSG:4326"
+        Coordinate Reference System for the GeoDataFrames.
+    tooltip : bool, optional, default=True
+        Whether to include tooltips (on hover) for the features.
+    popup : bool, optional, default=True
+        Whether to include popups (on click) for the features.
+
+    Returns
+    -------
+    folium.Map
+        A Folium map object with the PyPSA.Network components plotted.
+    """
+    if not explore_deps_present:
+        logger.warning(
+            "folium and mapclassify need to be installed to use `n.explore()`."
+        )
+        return None
+
+    from folium import FeatureGroup, LayerControl, Map, TileLayer
+
+    gdf_buses = gpd.GeoDataFrame(
+        n.buses, geometry=gpd.points_from_xy(n.buses.x, n.buses.y), crs=crs
+    )
+
+    n.links = n.links[n.links.carrier == "DC"]  # Filter out non DC links
```

Comment:
You are overwriting the network links here. This will break the network object. Use a temporary filtered list

Reply:
Good point, corrected!